gerritlab/global_vars.py

```diff
@@ -27,28 +31,143 @@ def load_config(remote, repo):
+        raise RuntimeError("Using private_token from .gitreview file is deprecated.")
+
+    raise SystemExit(f"Unable to find private token for {host}")
+
+
+class GitCredentialStore:
+    """
+    A class to interact with git credential store.
+    """
+    def __init__(self, host: str) -> None:
+        # TODO: use host + repo path to support per-repo tokens
+        self._host = host
+        self._token: Optional[str] = None
+        self._git_credentials: Dict[str, str] = {}
+        self._git_credentials_fetched: bool = False
+
+    def get_token(self) -> Union[str, None]:
+        """
+        Convenience method to get the token from the credential store.
+        """
+        return self.get("password")
+
+    def get(self, key: str, default=None) -> Union[str, None]:
+        """
+        Return key from internal git credentials cache, or default if not found.
+        """
+        if not self._git_credentials_fetched:
+            self._populate_git_credentials(self._call_git_credential_fill())
+            self._git_credentials_fetched = True
+
+        return self._git_credentials.get(key, default)
+
+    def _populate_git_credentials(self, git_credentials_output: str) -> None:
+        """
+        Populate the internal git credentials cache
+        """
+        for line in git_credentials_output.splitlines():
+            if "=" in line:
+                key, value = line.split("=", 1)
+                self._git_credentials[key] = value
+
+    def _call_git_credential_fill(self) -> str:
+        """
+        Look up credentials for the host using git credential fill.
+
+        Prepend 'gerritlab@' to ensure gitlab tokens can be stored separately
+        from other user tokens if desired.
+        """
+        try:
+            git_credentials = subprocess.run(
+                ["git", "credential", "fill"],
+                input=f"protocol=gerritlab\nhost=gerritlab@{self._host}\n\n",
+                text=True,
+                capture_output=True,
+            )
+        except KeyboardInterrupt:
+            return ""
+
+        if git_credentials.returncode != 0:
+            return ""
+        git_credential_output = git_credentials.stdout
+
+        # Store the credentials for later use
+        self._call_git_credential_approve(git_credential_output)
+        return git_credential_output
+
+    @staticmethod
+    def _call_git_credential_approve(git_credential_fill_out: str) -> int:
+        """
+        Attempt to save the credentials using git credential store.
+
+        Some credential-helpers (pass-git-helper) do not support saving. The
+        pattern seems to be to exit 0 without printing anything in that case.
+
+        Some credential-helpers print a helpful message to stderr.
+
+        Let's pass through the output to the user, but not treat as an error.
+        """
+        return subprocess.run(
+            ["git", "credential", "approve"],
+            input=git_credential_fill_out.encode("utf-8"),
```

Comment:
Consider using `text=True` instead of `.encode()` (easier to read and consistent with the code in `_call_git_credential_fill`)

Reply:
Done!